kafka-connect-jms/src/main/scala/com/datamountaineer/streamreactor/connect/jms/config/JMSSettings.scala

```diff
@@ -134,21 +136,30 @@ object JMSSettings extends StrictLogging {
     val jmsQueues = kcql.filter(k => k.getWithType.toUpperCase.equals("QUEUE")).map(k => if (sink) k.getTarget else k.getSource)
     val jmsSubscriptionName = config.getString(JMSConfigConstants.TOPIC_SUBSCRIPTION_NAME)
 
+    val headers = parseAdditionalHeaders(config.getString(s"${JMSConfigConstants.HEADERS_CONFIG}"))
+
     val settings = kcql.map(r => {
       val jmsName = if (sink) r.getTarget else r.getSource
       var converter = convertersMap.get(jmsName)
       if (converter.isEmpty) {
         converter = defaultConverter
       }
-
+      val headersForJmsDest = Splitter.on(',').omitEmptyStrings()
+        .split(headers.getOrElse(jmsName, ""))
+        .map { header =>
+          val keyValue = header.split(":", 2)
+          (keyValue(0), keyValue(1))
+        }
+        .toMap
```

Comment:
why not make a method to extract 'headersForJmsDest' directy ? 

you made a method to extract headers, but then you're doing additional post-processing of it here, in place, w/o using any external parameters

Reply:
Additional post-processing is JMS destination related. 
Maybe you're right, for additional post-processing I could create another method, but I don't think it's worth to perform extraction in a single method

Reply:
Nvm, I've just realized that it's nested within settings... :)

my bad